### DISCUSSION TITLE
关于工具微调

### DISCUSSION THREAD

**Discussion by yongdi1995** [zai-org/ChatGLM3#851] [Bad Case]
请问工具调用微调最近是关了吗
这个有计划支持吗
![](https://picgo-1305561115.cos.ap-beijing.myqcloud.com/img/20240221163657.png)

**Comment by zRzRzRzRzRzRzR** [comment]
可以按照原始代码改回去，不过效果挺差的。。

**Reply by tiamojames** [reply]
请问现在的代码还可以进行工具微调吗？

**Reply by FanZhang91** [reply]
原始代码在哪里？对应哪个分支？

**Comment by tiamojames** [comment]
> 可以按照原始代码改回去，不过效果挺差的。。
微调数据集格式里tool的设置这样子可以吗？
[
    {
      "tools" : [
        {
            "name": "track",
            "description": "追踪指定股票的实时价格",
            "parameters": {
                "type": "object",
                "properties": {
                    "symbol": {
                        "description": "需要追踪的股票代码"
                    }
                },
                "required": ['symbol']
            }
        },
        {
            "name": "text-to-speech",
            "description": "将文本转换为语音",
            "parameters": {
                "type": "object",
                "properties": {
                    "text": {
                        "description": "需要转换成语音的文本"
                    },
                    "voice": {
                        "description": "要使用的语音类型（男声、女声等）"
                    },
                    "speed": {
                        "description": "语音的速度（快、中等、慢等）"
                    }
                },
                "required": ['text']
            }
        }
    ],
      "conversations": [
        {
          "role": "user",
          "content": "<user prompt text>"
        },
        {
          "role": "assistant",
          "content": "<assistant thought to text>"
        },
        {
          "role": "tool",
          "name": "<name of the tool to be called",
          "parameters": {
            "<parameter_name>": "<parameter_value>"
          },
          "observation": "<observation>"
          // don't have to be string
        },
        {
          "role": "assistant",
          "content": "<assistant response to observation>"
        },
        // ... Muti Turn
        {
          "role": "user",
          "content": "<user prompt text>"
        },
        {
          "role": "assistant",
          "content": "<assistant response text>"
        }
      ]
    }
    // ...
  ]

**Reply by Admin-ppq** [reply]
你好，我可以问一下，原始代码在哪个分支下嘛？？？
